### PR TITLE
Migrate core/utils/coordinate.js to goog.module syntax

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from `goog_module`
- [x] My pull request is against `goog_module`
- [x] My code follows the [style guide](
      https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] My code is presented in the form suggested in the [module
      conversion guide](https://github.com/google/blockly/issues/5026)
- [x] I have run `npm test` locally already.

### Resolves

Part of https://github.com/google/blockly/issues/5026

### Proposed Changes

Converts `core/utils/coordinate.js` to `goog.module` syntax. 

### Test Coverage

Ran `npm run start` and `npm run test`

### Additional Information

No changes after running clang-format on this file.

### CODE REVIEW

File: core/utils/coordinate.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ Blockly.utils.Coordinate = function(x, y) {
  * @param {?Blockly.utils.Coordinate} b A Coordinate.
  * @return {boolean} True iff the coordinates are equal, or if both are null.
  */
-Blockly.utils.Coordinate.equals = function(a, b) {
+Coordinate.equals = function(a, b) {
```
Comment (maribethb) [inline core/utils/coordinate.js:49]: I think these (all the ones that aren't on the prototype) should be 
`const equals = function(a, b) {...}`

based on our discussion today, right? I think so, but this is the only one of the utils classes that mixes both prototype and non-prototype methods so I'm not sure.

File: core/utils/coordinate.js [hunk 2]
```diff
@@ -126,8 +126,10 @@ Blockly.utils.Coordinate.prototype.scale = function(s) {
...
  * @return {!Blockly.utils.Coordinate} This coordinate after translating.
  */
-Blockly.utils.Coordinate.prototype.translate = function(tx, ty) {
+Coordinate.prototype.translate = function(tx, ty) {
   this.x += tx;
   this.y += ty;
   return this;
 };
+
+exports = Coordinate;
```
Comment (maribethb) [inline core/utils/coordinate.js:135]: and the functions not on the prototype (if above comment applies)

File: core/utils/coordinate.js [hunk 3]
```diff
@@ -73,7 +73,7 @@ Blockly.utils.Coordinate.distance = function(a, b) {
  * @param {!Blockly.utils.Coordinate} a A Coordinate.
```
Comment (maribethb) [inline core/utils/coordinate.js:73]: Also, I think these references would all use just `Coordinate` now.

Comment (moniika) [reply core/utils/coordinate.js]: Done.

**Comment by moniika** [comment]
I've rebased so that I could fix the merge conflicts but still keep the commits organized.

**Review by maribethb — APPROVED** [review]

**Pull request merged**